posthog/temporal/data_imports/row_tracking.py

```diff
@@ -0,0 +1,62 @@
+from contextlib import contextmanager
+import uuid
+from posthog.exceptions_capture import capture_exception
+from posthog.redis import get_client
+
+
+def _get_hash_key(team_id: int) -> str:
+    return f"posthog:data_warehouse_row_tracking:{team_id}"
+
+
+@contextmanager
+def _get_redis():
+    try:
+        # Ensure redis is up and alive
+        redis = get_client()
+        redis.ping()
+
+        yield redis
+    except Exception as e:
+        capture_exception(e)
+
+
+def setup_row_tracking(team_id: int, schema_id: uuid.UUID | str) -> None:
+    with _get_redis() as redis:
+        redis.hset(_get_hash_key(team_id), str(schema_id), 0)
+        redis.expire(_get_hash_key(team_id), 60 * 60 * 24 * 7)  # 7 day expire
+
+
+def increment_rows(team_id: int, schema_id: uuid.UUID | str, rows: int) -> None:
+    with _get_redis() as redis:
+        redis.hincrby(_get_hash_key(team_id), str(schema_id), rows)
+
+
+def decrement_rows(team_id: int, schema_id: uuid.UUID | str, rows: int) -> None:
+    with _get_redis() as redis:
+        if not redis.hexists(_get_hash_key(team_id), str(schema_id)):
+            return
+
+        value = redis.hget(_get_hash_key(team_id), str(schema_id))
+        if not value:
+            return
+
+        value_int = int(value)
+        if value_int - rows < 0:
+            redis.hset(_get_hash_key(team_id), str(schema_id), 0)
```

Comment:
I guess we should never get here? Not sure if worth adding a log line to signify this?

Also, I guess strictly this should be wrapped in a transaction (the `hget` and `hset`) but I think this key should only be updated by this single process (unless it's possible to have multiple syncs for the same schema concurrently, which I don't think it is)

Reply:
Yeah, there's only a single process that can update this, so I'm not too fussed about keeping it atomic. Dont think we'll ever end up here, but I just wanna make sure we never go into negative values